core/src/Kokkos_Graph.hpp

```diff
@@ -39,14 +39,17 @@ namespace Experimental {
 //==============================================================================
 // <editor-fold desc="Graph"> {{{1
 
-template <class ExecutionSpace>
+template <class ExecutionSpace = DefaultExecutionSpace>
 struct [[nodiscard]] Graph {
+  static_assert(Kokkos::is_execution_space_v<ExecutionSpace>);
+
  public:
   //----------------------------------------------------------------------------
   // <editor-fold desc="public member types"> {{{2
 
   using execution_space = ExecutionSpace;
   using graph           = Graph;
+  using root_t          = GraphNodeRef<ExecutionSpace>;
```

Comment:
```suggestion
  using node_ref_root_t          = GraphNodeRef<ExecutionSpace>;
```

https://github.com/kokkos/kokkos/blob/13afdfd278ba9a56b5fbc3e32d9d90bd65980c62/core/src/impl/Kokkos_GraphNodeImpl.hpp#L46-L49

https://github.com/kokkos/kokkos/blob/13afdfd278ba9a56b5fbc3e32d9d90bd65980c62/core/unit_test/TestGraph.hpp#L810-L813

Reply:
I'm not reaaaallly in favor of such a verbose name. I'll do it if someone else also want it :wink:

Reply:
I think `root_t` is fine